libraries/chain/db_market.cpp

```diff
@@ -842,33 +840,30 @@ bool database::fill_call_order( const call_order_object& order, const asset& pay
             else if( get_dynamic_global_properties().next_maintenance_time > HARDFORK_CORE_343_TIME )
               o.call_price = price::call_price( o.get_debt(), o.get_collateral(),
                                 mia.bitasset_data(*this).current_feed.maintenance_collateral_ratio );
-       });
+      });
 
+   // update current supply
    const asset_dynamic_data_object& mia_ddo = mia.dynamic_asset_data_id(*this);
 
-   modify( mia_ddo, [&]( asset_dynamic_data_object& ao ){
-       //idump((receives));
-        ao.current_supply -= receives.amount;
+   modify( mia_ddo, [&receives]( asset_dynamic_data_object& ao ){
+         ao.current_supply -= receives.amount;
       });
 
-   const account_object& borrower = order.borrower(*this);
+   // adjust balance and/or account statistics
    if( collateral_freed.valid() || pays.asset_id == asset_id_type() )
```

Comment:
Add condition ``&& collateral_freed.amount != 0``?

Reply:
Actually the code is a bit messy here for potential better performance.
* The `OR` check here is for only updating the object once when possible;
* in most cases `collateral_freed.amount` is non-zero so it doesn't make much sense to add a check;
  * in addition, `adjust_balance()` will check if the delta is zero;
* actually there is a chance that we don't need to execute the `modify`, since actually `pays.asset_id` and `collateral_freed->asset_id` are the same, that means if we don't need to update `total_core_in_orders` due to `pays`, we won't need to update it due to `collateral_freed`. I will improve this. (Just saw your comment above)